src/community/ogcapi/ogcapi-tiles/src/main/java/org/geoserver/ogcapi/tiles/TileJSONBuilder.java

```diff
@@ -128,25 +128,33 @@ public TileJSON build() throws FactoryException, TransformException, IOException
         String tilesURL;
         APIRequestInfo requestInfo = APIRequestInfo.get();
         String baseURL = requestInfo.getBaseURL();
+        boolean isVector =
+                styleId == null
+                        && "application/vnd.mapbox-vector-tile".equalsIgnoreCase(tileFormat);
```

Comment:
There are other formats to represent a vector tile (geojson, topojson).
You can use the the GWC class ``MimeType.createFromFormat(tileFormat)`` and then call ``isVector`` to find out if a given mime type is meant for vector tiles.

Reply:
Right. fix on its way.